src/iexamine.cpp

```diff
@@ -2979,23 +2996,14 @@ void iexamine::tree_maple_tapped( player &p, const tripoint &examp )
             return;
         }
 
-        case HARVEST_SAP:
-            for( auto &it : items ) {
-                if( ( it.is_bucket() || it.is_watertight_container() ) && !it.is_container_empty() ) {
-                    auto &liquid = it.contents.front();
-                    if( liquid.typeId() == "maple_sap" ) {
-                        liquid_handler::handle_liquid_from_container( it, PICKUP_RANGE );
-                    }
-                }
-            }
-
+        case HARVEST_SAP: {
+            liquid_handler::handle_liquid_from_container( *container, PICKUP_RANGE );
```

Comment:
Looks like `*container` can be a `nullptr` and there is no check for that.

Reply:
This option is only available if `has_sap` is true, and `has_sap` can only be true if a container was found.

Reply:
I'm not saying this code isn't fragile and prone to break at some point, but it is currently correct I believe.